### DISCUSSION TITLE
Is it possible to limit RAM/memory usage and prevent OOM?

### DISCUSSION THREAD

**Discussion by sheyt0** [etcd-io/etcd#18031] [Q&A]
With overactive read requests for large range requests, members of my cluster may get an OOM.
Which ultimately leads to cluster instability due to the sequential death of all members of the cluster.

I tried limiting memory usage using MemoryHigh (soft limit) for the etcd systemd unit, but this caused the member to stop responding to any requests, including stats requests, when the limit is reached.

Is there a way to limit memory usage in order to preserve the life of the cluster at the cost of query processing degradation?

**Comment by ahrtr** [comment]
> With overactive read requests for large range requests, members of my cluster may get an OOM

Please try to avoid large range requests. OOM is a known issue.
